### PR TITLE
Add --fatal=errors|warnings program option

### PR DESCRIPTION
**Pull request by kernc**
Or is there an alternative when one is using continuous integration to build and relies on program exiting cleanly (or not) as an indicator of success?

### CODE REVIEW

File: pelican/log.py [hunk 1]
```diff
@@ -203,7 +223,10 @@ def get_formatter():
         return TextFormatter()
 
 
-def init(level=None, handler=logging.StreamHandler(), name=None):
+def init(level=None, warnings_fatal=False, errors_fatal=False,
+         handler=logging.StreamHandler(), name=None):
+    FatalLogger.warnings_fatal = warnings_fatal
+    FatalLogger.errors_fatal = errors_fatal or warnings_fatal
```
Comment (avaris) [inline pelican/log.py]: Isn't this backwards? `warnings_fatal` should exit if warnings or errors occur, not the other way around.

**Comment by avaris** [comment]
Also, can we not spread `sys.exit` all around the code? Multiple exit points just makes maintaining harder. Just raise an exception and let the code in `__init__.py` handle it.

File: pelican/log.py (hunk 1 above)
Comment (kernc) [inline pelican/log.py]: After reading your email yesterday, I was actually agreeing with your observation right up until this moment when I saw the edit. These predicates be hard! :laughing:

**Comment by justinmayer** [comment]
Hey @kernc! Many thanks for the follow-up changes. It looks like there is one [PEP-8 test failure](https://travis-ci.org/getpelican/pelican/jobs/91681877) in 86b7109 that needs addressing. Would you mind fixing it and amending that commit?

@avaris: Any other changes needed from your perspective?

**Comment by kernc** [comment]
Amended, thanks.

I'm not completely completely sold on `--errors-fatal` switch name, though. Does `--fail-on` or even just `--fatal errors|warnings` sound better and acceptable?

**Comment by kernc** [comment]
Have settled with `--fatal`.

**Comment by justinmayer** [comment]
Nice work, @kernc. Many thanks for your efforts on this feature!

**Comment by ingwinlu** [comment]
This breaks pelican when --fatal is not set.

https://gist.githubusercontent.com/ingwinlu/f39d5b6345f576cd2893/raw/6f6e02278b55fc1cd2b18176665d41a84152c24e/pelican%2520master%2520log,%2520.fatal%2520=%2520None

Examining args shows that obviously fatal defaults to None when it is not set resulting in the error above
```
Namespace(autoreload=False, cache_path=None, delete_outputdir=None, fatal=None, ignore_cache=False, output='/home/winlu/gitrepos/www.heroicdebugging.biz/output', path='/home/winlu/gitrepos/www.heroicdebugging.biz/content', relative_paths=False, selected_paths=None, settings='/home/winlu/gitrepos/www.heroicdebugging.biz/pelicanconf.py', theme=None, verbosity=10)
```

**Comment by kernc** [comment]
Har, har, hurr!  :sweat_smile:

**Pull request merged**